cdap-data-fabric/src/main/java/co/cask/cdap/data2/security/Impersonator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.security;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.kerberos.SecurityUtil;
+import co.cask.cdap.proto.id.NamespaceId;
+import com.google.inject.Inject;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Responsible for executing code for a user, configurable at the namespace level.
+ */
+public class Impersonator {
+
+  private final boolean kerberosEnabled;
+  private final UGIProvider ugiProvider;
+  private final ImpersonationUserResolver impersonationUserResolver;
+
+  @Inject
+  public Impersonator(CConfiguration cConf, UGIProvider ugiProvider,
```

Comment:
no, I didn't mean you should :) I did some of it in my PR. I was just trying to find out why this needs to be public. Perhaps add a ``@VisibleForTesting`` with a comment. Other than that, no action required :)